### ISSUE TITLE
Make file_cache_path configurable

### ISSUE THREAD

**Issue by benthorner** [matschaffer/knife-solo#508]
Currently the local-mode-cache directory is hard-coded to be under the 'provisioner-path', which is normally the user's home directory. This causes issues for a team environment: multiple copies of cached files; unnecessary re-installations of cached binaries.

Changing the whole 'provisioner-path' could cause permission issues, but can we make the local-mode-cache directory configurable? It's owned by root and only accessed by chef-client, so there shouldn't be any permission problems from (optionally) moving it to a global system location.

**Comment by tpbowden** [comment]
Yes I agree this would be useful for deploying across multiple user accounts.

**Comment by tpbowden** [comment]
Would it be possible to read this from `knife.rb` and then write it to `knife-solo/lib/knife-solo/resources/solo.rb.erb` like how proxies are set?

**Comment by matschaffer** [comment]
Seems like a reasonable thing to make configurable. Especially given some precident.
